arch/risc-v/src/mpfs/mpfs_opensbi_utils.S

```diff
@@ -83,6 +70,22 @@ mpfs_global_pointer:
 
   .align 3
 mpfs_opensbi_prepare_hart:
+  csrr  a0, mhartid
+  li    a1, 3       /* hart3 hardcoded */
```

Comment:
This looks fishy to me, why hart 3? Could the copying be done by the bootloader hart (config flag, typically hart 0 for us...)

Reply:
Hart0 doesn't run here, the copy function would need to take place at the main.c/bl.c?

Reply:
Made it a function now, so may be called from anywhere